components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/federatedIdpInitLogout/factory/LogoutRequestFactory.java

```diff
@@ -0,0 +1,80 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authenticator.oidc.federatedIdpInitLogout.factory;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.authentication.framework.inbound.FrameworkClientException;
+import org.wso2.carbon.identity.application.authentication.framework.inbound.HttpIdentityRequestFactory;
+import org.wso2.carbon.identity.application.authentication.framework.inbound.IdentityRequest;
+import org.wso2.carbon.identity.application.authenticator.oidc.model.LogoutRequest;
+
+import java.util.regex.Matcher;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import static org.wso2.carbon.identity.application.authenticator.oidc.OIDCAuthenticatorConstants.OIDC_BACKCHANNEL_LOGOUT_ENDPOINT_URL_PATTERN;
+
+/**
+ * Checks whether requests from the Identity servlet are OIDC Logout Requests and
+ * provides a builder to an instance of OIDC LogoutRequest.
+ */
+public class LogoutRequestFactory extends HttpIdentityRequestFactory {
+
+    private static final Log log = LogFactory.getLog(LogoutRequestFactory.class);
+
+    @Override
+    public boolean canHandle(HttpServletRequest request, HttpServletResponse response) {
+
+        boolean canHandle = false;
+        Matcher registerMatcher =
+                OIDC_BACKCHANNEL_LOGOUT_ENDPOINT_URL_PATTERN.matcher(request.getRequestURI());
+        if (registerMatcher.matches()) {
+            canHandle = true;
+        }
+        if (log.isDebugEnabled()) {
+            log.debug("canHandle " + canHandle + " by OIDC LogoutRequestFactory.");
+        }
+        return canHandle;
+    }
+
+    @Override
+    public LogoutRequest.LogoutRequestBuilder create(HttpServletRequest request,
+                                                     HttpServletResponse response)
+            throws FrameworkClientException {
+
+        LogoutRequest.LogoutRequestBuilder logoutRequestBuilder = new LogoutRequest.
+                LogoutRequestBuilder(request, response);
+        create(logoutRequestBuilder, request, response);
+        return logoutRequestBuilder;
+
+    }
+
+    @Override
+    public void create(IdentityRequest.IdentityRequestBuilder builder, HttpServletRequest request,
+                       HttpServletResponse response) throws FrameworkClientException {
+
+        LogoutRequest.LogoutRequestBuilder registerRequestBuilder;
+        if (builder instanceof LogoutRequest.LogoutRequestBuilder) {
```

Comment:
Do we need this instance check and cast?
What if u simply no override this method?